### PR TITLE
Refs #27683 -- Allowed setting isolation level in DATABASES ['OPTIONS'] on MySQL.

### PR DESCRIPTION
**Pull request by timgraham**
Updated from https://github.com/django/django/pull/7826.

### CODE REVIEW

File: django/db/backends/mysql/base.py [hunk 1]
```diff
@@ -262,13 +284,20 @@ def get_new_connection(self, conn_params):
...
     def init_connection_state(self):
+        assignments = []
         if self.features.is_sql_auto_is_null_enabled:
+            # SQL_AUTO_IS_NULL controls whether an AUTO_INCREMENT column on
+            # a recently inserted row will return when the field is tested
+            # for NULL. Disabling this brings this aspect of MySQL in line
+            # with SQL standards.
+            assignments.append('SQL_AUTO_IS_NULL = 0')
+
+        if self.isolation_level_value:
```
Comment (shaib) [inline django/db/backends/mysql/base.py]: This attribute does not exist if `isolation_level` has not been specified in `OPTIONS`, AFAICT

File: django/db/backends/mysql/base.py [hunk 2]
```diff
@@ -252,7 +259,23 @@ def get_connection_params(self):
         # We need the number of potentially affected rows after an
         # "UPDATE", not the number of changed rows.
         kwargs['client_flag'] = CLIENT.FOUND_ROWS
-        kwargs.update(settings_dict['OPTIONS'])
+        # Validate the transaction isolation level, if specified.
+        options = settings_dict['OPTIONS'].copy()
+        isolation_level = options.pop('isolation_level', None)
+        if isolation_level:
+            isolation_level = isolation_level.lower()
```
Comment (timgraham) [inline django/db/backends/mysql/base.py]: Should we add a test for accepting upper cased values or remove this line?

Comment (shaib) [inline django/db/backends/mysql/base.py]: add a test, they show up upper cased in all the SQL documentations.

**Comment by shaib** [comment]
Thanks! I was too busy all day yesterday to do more than comment...

**Pull request merged**